File: MAP/Csharp lab2/Csharp lab2/Ui/Validator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Csharp_lab2.Domain;

namespace Csharp_lab2.Ui
{
    class Validator
    {
        public bool validateGrade(int g)
        {
            if (g < 1 || g > 10)
                throw new MyException("Invalid grade");
            return true;
        }

        public bool validateName(string name)
        {
            if (name == "")
                throw new MyException("Empty string for name");
            return true;
        }

        public bool validateId(int id)
        {
            if (id < 1)
                throw new MyException("Id cannot be below 1");
            return true;
        }
    }
}

File: MAP/Lab7MAP/src/controller/Validator.java
package controller;
import domain.MyException;
public class Validator {
	
	public boolean validateGrade(int g) throws MyException{
		if(g<1||g>10)
			throw new MyException("Invalid grade");
		return true;
	}
	
	public boolean validateName(String name) throws MyException{
		if(name=="")
			throw new MyException("Empty string for name");
		return true;
	}
	
	public boolean validateId(int id) throws MyException{
		if(id<1)
			throw new MyException("Id must be greater than 0");
		return true;
	}
	

}
